nyaa/views/account.py

```diff
@@ -127,6 +127,27 @@ def profile():
     return flask.render_template('profile.html', form=form)
 
 
+@bp.route('/notifications', endpoint='notifications', methods=['GET'])
+def view_notifications():
+    if not flask.g.user:
+        flask.abort(403)
+
+    page = flask.request.args.get('p', flask.request.args.get('offset', 1, int), int)
+    notifications = models.Notification.get_notifications(flask.g.user.id, page)
+    # Users unable to see which are read and unread (bolded)
+    # updates = models.Notification.mark_notifications_read(flask.g.user.id)
```

Comment:
Have you tried rendering the template before calling `mark_notifications_read`?
I'm pretty sure that because you're altering the database, the notifications are marked as read before you actually render the template.
Try this:
```python
# [...]
notifications = models.Notification.get_notifications(flask.g.user.id, page)
template = flask.render_template('notifications.html', notifications=notifications)

updates = models.Notification.mark_notifications_read(flask.g.user.id)
db.session.commit()

return template
```